sarathi/model_executor/attention/base_attention_wrapper.py

```diff
@@ -37,6 +38,32 @@ def get_timer(self, operation: OperationMetrics, layer_id: Optional[int] = None)
         if self._timers.get((operation, layer_id)) is None:
             self._timers[(operation, layer_id)] = CudaTimer(operation, layer_id)
         return self._timers.get((operation, layer_id))
+    
+    def init_gpu_cache(self, num_gpu_blocks: int) -> None:
+        gpu_cache: List[torch.Tensor] = []
+        self.num_gpu_blocks = num_gpu_blocks
+
+        for _ in range(self.num_layers):
+            gpu_blocks = self.get_cache_block(
+                self.num_gpu_blocks, dtype=self.dtype, device="cuda"
+            )
+            gpu_cache.append(gpu_blocks)
+        
+        self.gpu_cache = gpu_cache
+
+    def get_cache_block(self, num_blocks: int, **kwargs) -> torch.Tensor:
+        return torch.randn(
```

Comment:
this function should remain within the child class? different backends could have different block table formats?

Reply:
Makes sense.

This was moved to the base class as per this [comment](https://github.com/microsoft/sarathi-serve/pull/32#discussion_r1694123209).

Reply:
Moved it out to child class